WalletWasabi.Fluent/Behaviors/MainWindowPersistPositionBehavior.cs

```diff
@@ -0,0 +1,97 @@
+using Avalonia;
+using Avalonia.Controls;
+using ReactiveUI;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using WalletWasabi.Fluent.ViewModels;
+using System.Linq;
+
+namespace WalletWasabi.Fluent.Behaviors;
+
+public class MainWindowBindPositionBehavior : DisposingBehavior<Window>
+{
+	protected override void OnAttached(CompositeDisposable disposables)
+	{
+		if (AssociatedObject is null)
+		{
+			return;
+		}
+
+		Observable.FromEventPattern(AssociatedObject, nameof(Window.DataContextChanged))
+				  .Subscribe(_ => BindPosition(disposables))
+				  .DisposeWith(disposables);
+	}
+
+	private void BindPosition(CompositeDisposable disposables)
+	{
+		if (AssociatedObject?.DataContext is MainViewModel vm)
+		{
+			Observable.FromEventPattern(AssociatedObject, nameof(Window.PositionChanged))
+				  .Subscribe(_ => vm.WindowPosition = AssociatedObject.Position)
+				  .DisposeWith(disposables);
+
+			vm.WhenAnyValue(x => x.WindowPosition)
+			  .Where(x => x is { })
+			  .Subscribe(x => AssociatedObject.Position = x.Value)
+			  .DisposeWith(disposables);
+
+			SetInitialPosition(vm);
+		}
+	}
+
+	private void SetInitialPosition(MainViewModel vm)
+	{
+		if (vm.WindowState != WindowState.Normal)
+		{
+			return;
+		}
+
+		var centerX = (int)((AssociatedObject!.Screens.Primary.Bounds.Width - vm.WindowWidth) / 2);
+		var centerY = (int)((AssociatedObject!.Screens.Primary.Bounds.Height - vm.WindowHeight) / 2);
```

Comment:
I wonder if the `WorkingArea` would make more sense instead of the `Bounds`. So the taskbar wouldn't be taken into account and the window would be truly centralized.

Reply:
@soosr done